raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientReconfigTest.java

```diff
@@ -2817,9 +2830,16 @@ void testUpdateVoterResponseCausesEpochChange() throws Exception {
             )
         );
 
-        // check that there is a fetch to the new leader
+        // the first poll can still send a fetch request to the old leader, because there is not one in flight
+        // but will handle the update voter response afterwards to update state
         context.pollUntilRequest();
         RaftRequest.Outbound fetchRequest = context.assertSentFetchRequest();
+        context.assertFetchRequestData(fetchRequest, epoch, 0L, 0);
+        assertEquals(voter1.id(), fetchRequest.destination().id());
```

Comment:
Interesting behavior. We should add a test that this FETCH response is ignored because the UpdateVoterResponse above has a high epoch which forces the local replica to transition. We can make this check here since this test is already trying to test what happens during an epoch change.